nxt_editor/stage_model.py

```diff
@@ -2169,6 +2170,138 @@ def _remove_breakpoint(self, node_path, layer):
+        on = []
+        off = []
+        for node_path in node_paths:
+            current_state = self.is_node_skippoint(node_path, layer_path)
+            node_val = not current_state
+            if node_val:
+                on += [node_path]
+            else:
+                off += [node_path]
+        node_count = len(node_paths)
+        if node_count > 1:
+            msg = ('Set skippoint for {} and '
+                   '{} other(s)'.format(node_paths[0], node_count - 1))
+        else:
+            msg = 'Set skippoint for {}'.format(node_paths[0])
+        self.undo_stack.beginMacro(msg)
+        if on:
+            self.set_skippoints(on, True, layer_path)
+        if off:
+            self.set_skippoints(off, False, layer_path)
+        self.undo_stack.endMacro()
+
+    def toggle_descendant_skips(self, node_paths, layer_path=None):
+        """Reverse the skip point status of each node given, applying the same
+        status to all descendant nodes.
+
+        :param node_paths: Root node(s) to change skip status of.
+        :type node_paths: iterable
+        :param layer_path: Layer to set skips on, defaults to top layer.
+        :type layer_path: str, optional
+        """
+        node_count = len(node_paths)
+        if node_count > 1:
+            msg = ('Set skippoint for {} and '
+                   '{} other(s)'.format(node_paths[0], node_count - 1))
+        else:
+            msg = 'Set skippoint for {}'.format(node_paths[0])
+        self.undo_stack.beginMacro(msg)
+        for node_path in node_paths:
+            to_skip = not self.is_node_skippoint(node_path)
+            descendants = self.get_descendants(node_path)
+            self.set_skippoints(descendants + [node_path], to_skip, layer_path)
+        self.undo_stack.endMacro()
+
+    def set_skippoints(self, node_paths, to_skip, layer_path=None):
+        """Set the skip status of given nodes.
+
+        :param node_paths: Nodes to set skip status of.
+        :type node_paths: iterable
+        :param to_skip: Whether to set nodes to skip or not.
+        :type to_skip: bool
+        :param layer_path: Layer to set skips on, defaults to top layer.
+        :type layer_path: str, optional
+        """
+        if not layer_path:
+            layer_path = self.top_layer.real_path
+        cmd = SetNodesAreSkipPoints(node_paths, to_skip, layer_path, self)
+        self.undo_stack.push(cmd)
+
+    def is_node_skippoint(self, node_path, layer_path=None):
+        """Returns True/False based on whether a node is currently a skippoint.
+
+        :param node_path: Node to check.
+        :type node_path: str
+        :param layer_path: Layer path to check within, defaults to top layer.
+        :type layer_path: str, optional
+        :return: Whether given node is a skip.
+        :rtype: bool
+        """
+        if not layer_path:
+            layer_path = self.top_layer.real_path
+        layer_skips = user_dir.skippoints.get(layer_path, tuple())
```

Comment:
Everywhere else it seems to be a list.
```suggestion
        layer_skips = user_dir.skippoints.get(layer_path, [])
```